java/vector/src/test/java/org/apache/arrow/vector/file/BaseFileTest.java

```diff
@@ -541,4 +542,52 @@ public void writeUnionData(int count, NullableMapVector parent) {
     writer.setValueCount(count);
     varchar.release();
   }
+
+  protected void writeVarBinaryData(int count, NullableMapVector parent) {
+    Assert.assertTrue(count < 100);
+    ComplexWriter writer = new ComplexWriterImpl("root", parent);
+    MapWriter rootWriter = writer.rootAsMap();
+    ListWriter listWriter = rootWriter.list("list");
+    ArrowBuf varbin = allocator.buffer(count);
+    for (int i = 0; i < count; i++) {
+      varbin.setByte(i, i);
+      listWriter.setPosition(i);
+      listWriter.startList();
+      for (int j = 0; j < i % 3; j++) {
+        listWriter.varBinary().writeVarBinary(0, i + 1, varbin);
+      }
+      listWriter.endList();
+    }
+    writer.setValueCount(count);
+    varbin.release();
+  }
+
+  protected void validateVarBinary(int count, VectorSchemaRoot root) {
+    Assert.assertEquals(count, root.getRowCount());
+    ListVector listVector = (ListVector) root.getVector("list");
+    byte[] expectedArray = new byte[count];
+    int numVarBinaryValues = 0;
+    for (int i = 0; i < count; i++) {
+      expectedArray[i] = (byte) i;
+      Object obj = listVector.getAccessor().getObject(i);
+      List<?> objList = (List) obj;
+      if (i % 3 == 0) {
+        Assert.assertTrue(objList.isEmpty());
+      } else {
+        byte[] expected = Arrays.copyOfRange(expectedArray, 0, i + 1);
+        for (int j = 0; j < i % 3; j++) {
+          byte[] result = (byte[]) objList.get(j);
+          Assert.assertArrayEquals(result, expected);
+          numVarBinaryValues++;
+        }
+      }
+    }
+
+    // ListVector lastSet should be the index of last value + 1
+    Assert.assertEquals(listVector.getMutator().getLastSet(), count);
+
+    // NullableVarBinaryVector lastSet should be the index of last value
+    NullableVarBinaryVector binaryVector = (NullableVarBinaryVector) listVector.getChildrenFromFields().get(0);
+    Assert.assertEquals(binaryVector.getMutator().getLastSet(), numVarBinaryValues - 1);
```

Comment:
@icexelloss I added tests for ListVector with NullableVarBinaryVector that also checks `lastSet` values.  Please take another look when you get a chance, thanks!

Reply:
I thought the tests were already there covering this scenario in TestListVector.java. I remember adding tests for setLastSet().

Reply:
@siddharthteotia I think the issue is TestListVector doesn't test the case that a ListVector is created from json reader.